src/content/docs/en/reference/api-reference.mdx

```diff
@@ -778,10 +826,79 @@ const draftBlogPosts = await getCollection('blog', ({ data }) => {
 
 [See the `Content Collection` guide](/en/guides/content-collections/#querying-collections) for example usage.
 
+### `getEntry()`
+
+<p>
+  <Since v="2.5.0" />
+</p>
+
+**Types:**
+
+- `(collection: string, contentSlugOrDataId: string) => CollectionEntry<collection>`
+- `({ collection: string, id: string }) => CollectionEntry<collection>`
+- `({ collection: string, slug: string }) => CollectionEntry<collection>`
+
+`getEntry()` is a function that retrieves a single collection entry by collection name and either the entry `id` (for JSON or YAML collections) or entry `slug` (for Markdown or MDX collections).
+
+```astro
+---
+import { getEntry } from 'astro:content';
+
+// Get `src/content/blog/enterprise.md`
+const enterprisePost = await getEntry('blog', 'enterprise');
+
+// Get `src/content/captains/captain-kirk.yaml`
+const captainProfile = await getEntry('captains', 'captain-kirk');
+---
+```
+
+`getEntry()` can also be used to get referenced entries to access the `data`, `body`, or `render()` properties:
+
+```astro
+---
+import { getEntry } from 'astro:content';
+
+const enterprisePost = await getEntry('blog', 'enterprise');
+
+// Get the profile referenced by `data.captain`
+const captainProfile = await getEntry(enterprise.data.captain);
+---
+```
+
+See the `Content Collection` guide for examples of [querying](/en/guides/content-collections/#querying-collections) and [references](/en/guides/content-collections/#resolving-references).
+
+### `getEntries()`
+
+<p>
+  <Since v="2.5" />
+</p>
+
+**Types:**
+
+- `(Array<{ collection: string, id: string }>) => CollectionEntry<collection>`
+- `(Array<{ collection: string, slug: string }>) => CollectionEntry<collection>`
+
+`getEntries()` is a function that retrieves multiple collection entries with the same collection name. This is useful for resolving an array of [references](en/guides/content-collections/#defining-collection-references) to their associated `data`, `body`, and `render()` properties.
+
+```astro
+---
+import { getEntries } from 'astro:content';
+
+const enterprisePost = await getEntry('blog', 'enterprise');
+
+// Get related posts referenced by `data.relatedPosts`
+const captainProfile = await getEntries(enterprise.data.relatedPosts);
+---
+```
+
 ### `getEntryBySlug()`
```

Comment:
Can we... deprecate `getEntryBySlug`? :sweat_smile:  I mean, *say that* it's deprecated/legacy? (Even if we don't actually deprecate for a loooong time.) I feel like it's just messy to keep this around, and define/explain it if another function entirely replaces it. It would be cleaner to just say update, because this will eventually go away.

Reply:
I'm trying to remember/hunt down what we did for deprecation in the past. I feel like we actually used the `<Badge /> component or something like that for emphasis. I think updating the note for now should be fine, though!

Reply:
Looks like we use `:::caution[Deprecated]` for `Astro.canonicalURL`. Think that's a good way forward!